packages/spec/src/runner.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright IBM Corp. 2019, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @typedef Runner
+ * @property {Function} beforeEach
+ * @property {Function} afterEach
+ * @property {Function} run
+ */
+
+/**
+ * Create a runner for the given collection of rules
+ * @param {Array<Rule>} rules
+ * @returns Runner
+ */
+export function createRunner(rules, options = {}) {
+  const { globals, only = [], exclude = [] } = options;
+  const levels = {
+    error: 'violation',
+    warning: 'warning',
+  };
+  const rulesToRun = rules.filter(rule => {
+    if (only.length > 0) {
+      return only.includes(rule.id);
+    } else if (exclude.length > 0) {
+      return !exclude.includes(rule.id);
+    }
+    // By default, run all rules
+    return true;
+  });
+
+  let _beforeAll;
+  let _beforeEach;
+  let _afterEach;
+
+  return {
+    beforeAll(fn) {
+      _beforeAll = fn;
+    },
+    beforeEach(fn) {
+      _beforeEach = fn;
+    },
+    afterEach(fn) {
+      _afterEach = fn;
+    },
+    test() {
+      const { describe, beforeAll, beforeEach, afterEach, test } = globals;
+
+      rulesToRun.forEach(rule => {
+        describe(rule.id, () => {
+          let node;
+
+          beforeAll(() => {
+            _beforeAll();
+          });
+
+          beforeEach(() => {
+            node = _beforeEach(rule.context);
```

Comment:
It would be awesome if we could make this portion of the test runner more flexible. I know the way karma is setup, we'd want to have different parameters passed in the `beforeEach` function and although this file is considered framework specific, it'd be nice to work together and see if there is a way to have it more flexible so we aren't having to create multiple files with only a few lines of differences.

Reply:
Agreed! This was only setup for the purposes of integrating into Jest. Like the PR comments suggest, almost all of this should be changed 😄